react-basic-practice/src/pages/ProductDetail/index.jsx

```diff
@@ -59,19 +59,29 @@ const ProductDetail = () => {
 
   const handleAddCart = async () => {
     try {
-      const cartUser = {
-        id: userData.userId,
-        listProducts: [
-          ...listItem.listProducts,
-          {
-            idProduct: product.id,
-            quantity: quantityProduct,
-            name: product.name,
-            price: product.price,
-          }],
-      };
+      let cartUser = {};
+      const haveProduct = cart?.products?.findIndex(
```

Comment:
Use `array.some` instead of `findIndex`.

Reply:
I need the position of the item to update in the array so use `findIndex`

Reply:
Then you must rename the variable.